src/utils.rs

```diff
@@ -0,0 +1,70 @@
+use std::path::{Path, PathBuf};
+use std::io::Error as IoError;
+use std::env::VarError;
+
+#[derive(Debug)]
+pub enum Error {
+    EnvError(VarError),
+    IoError(IoError),
+}
+
+impl From<VarError> for Error {
+    fn from(e: VarError) -> Self {
+        Error::EnvError(e)
+    }
+}
+
+impl From<IoError> for Error {
+    fn from(e: IoError) -> Self {
+        Error::IoError(e)
+    }
+}
+
+fn not_found_error() -> IoError {
+    use std::io::ErrorKind as IoErrorKind;
+    IoError::new(IoErrorKind::NotFound, "No file name component")
+}
+
+#[allow(dead_code)]
```

Comment:
I would say we can add a platform specific condition in that case `[fg(if not platofrm) "linux"]` type condition